lib/active_merchant/billing/gateways/netbanx.rb

```diff
@@ -0,0 +1,245 @@
+        end
+      end
+
+      def expdate(credit_card)
+        year  = format(credit_card.year, :four_digits)
+        month = format(credit_card.month, :two_digits)
+
+        # returns a hash (necessary in the card JSON object)
+        { :month => month, :year => year }
+      end
+
+      def add_order_id(post, options)
+        post[:merchantRefNum] = (options[:order_id] || SecureRandom.uuid)
+      end
+
+      def map_address(address)
+        return {} if address.nil?
+        country = Country.find(address[:country]) if address[:country]
+        mapped = {
+          :street  => address[:address1],
+          :city    => address[:city],
+          :zip     => address[:zip],
+        }
+        mapped.merge!({:country => country.code(:alpha2).value}) unless country.blank?
+
+        mapped
+      end
+
+      def parse(body)
+        body.blank? ? {} : JSON.parse(body)
+      end
+
+      def commit(method, uri, parameters)
+        params = parameters.to_json unless parameters.nil?
+        response = begin
+          parse(ssl_request(method, get_url(uri), params, headers))
+        rescue ResponseError => e
+          return Response.new(false, 'Invalid Login') if(e.response.code == '401')
+          parse(e.response.body)
+        end
+
+        success = success_from(response)
+        Response.new(
+          success,
+          message_from(success, response),
+          response,
+          :test => test?,
+          :authorization => authorization_from(success, get_url(uri), method, response)
+        )
+      end
+
+      def get_url(uri)
+        url = (test? ? test_url : live_url)
+        if uri =~ /^customervault/
+          "#{url}#{uri}"
+        else
+          "#{url}cardpayments/v1/accounts/#{@options[:account_number]}/#{uri}"
+        end
+      end
+
+      def success_from(response)
+        response.blank? || !response.key?('error')
+      end
+
+      def message_from(success, response)
+        success ? 'OK' : (response['error']['message'] || "Unknown error - please contact Netbanx-Paysafe")
+      end
+
+      def authorization_from(success, url, method, response)
+        if success && response.present? && url.match(/cardpayments\/v1\/accounts\/.*\//)
+          response['id']
+        elsif method == :post && url.match(/customervault\/.*\//)
+          # auth for tokenised customer vault is returned as
+          # customer_profile_id|card_id|payment_method_token
+          #
+          # customer_profile_id is the uuid that identifies the customer
+          # card_id is the uuid that identifies the card
+          # payment_method_token is the token that needs to be used when
+          #                      calling purchase with a token
+          #
+          # both id's are used to unstore, the payment token is only used for
+          # purchase transactions
+          [response['id'], response['cards'].first['id'], response['cards'].first['paymentToken']].join("|")
```

Comment:
Sure. When you try to unstore a customer or a card, you need to use the `id` however when you want to make a payment using a stored card, you have to use the `paymentToken` so I thought it would be good to pass all of them back. The `customer_id` or `card_id` are also used in the backend administrative interface to search.